test/Microsoft.AspNet.Mvc.TagHelpers.Test/TestModelMetadataProvider.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.ModelBinding.Metadata;
+using Microsoft.Framework.Internal;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public class TestModelMetadataProvider : DefaultModelMetadataProvider
```

Comment:
I'm just going to add a TestCommon in a follow up. We have a bunch of things like this already and that will scale better.